tests/test_dataclasses.py

```diff
@@ -380,3 +380,28 @@ class Outer:
             }
         },
     }
+
+
+def test_initvar():
+    from math import pi
+
+    InitVar = dataclasses.InitVar
+
+    @pydantic.dataclasses.dataclass
+    class RadiansWithInitVar:
+        rad: float = dataclasses.field(init=False)
+        deg: InitVar
+
+        def __post_init__(self, deg):
+            if deg:
+                self.rad = float(deg) / 180 * pi
```

Comment:
perhaps easier not to do anything this clever in a test for `InitVar`, instead just assert that deg is what you expect.

Reply:
My use case for InitVar is to derive multiple model fields from a complex type. I wanted to test that this derivation works with pydantic dataclasses.

Reply:
ok thats fine, but we're running here before we can walk, please add a first test that demonstrates the simplest possible use of `InitVar`.

Also, we're not testing maths here, the test can just add 1 here and still test the required functionality.

Unit tests should try to be as simple as possible while testing specific functionality, anything unnecessary (eg. calculations like) just make the test harder to read debug.